pep-0723.rst

```diff
@@ -73,24 +73,39 @@ begun getting frustrated with the lack of unification regarding both tooling
 and specs. Adding yet another way to define metadata, even for a currently
 unsatisfied use case, would further fragment the community.
 
-A use case that this PEP wishes to support that other formats may preclude is
-a script that desires to transition to a directory-type project. A user may
-be rapidly prototyping locally or in a remote REPL environment and then decide
-to transition to a more formal project if their idea works out. This
-intermediate script stage would be very useful to have fully reproducible bug
-reports. By using the same metadata format, the user can simply copy and paste
-the metadata into a ``pyproject.toml`` file and continue working without having
-to learn a new format. More likely, even, is that tooling will eventually
-support this transformation with a single command.
+The following are some of the use cases that this PEP wishes to support:
+
+* A user facing CLI that is capable of executing scripts. If we take Hatch as
+  an example, the interface would be simply
+  ``hatch run /path/to/script.py [args]`` and Hatch will manage the
+  environment for that script. Such tools could be used as shebang lines on
+  non-Windows systems e.g. ``#!/usr/bin/env hatch run``. You would also be
+  able to enter a shell into that environment like other projects by doing
+  ``hatch -p /path/to/script.py shell`` since the project flag would learn
+  that project metadata could be read from a single file.
+* A script that desires to transition to a directory-type project. A user may
+  be rapidly prototyping locally or in a remote REPL environment and then
+  decide to transition to a more formal project layout if their idea works
+  out. This intermediate script stage would be very useful to have fully
+  reproducible bug reports. By using the same metadata format, the user can
+  simply copy and paste the metadata into a ``pyproject.toml`` file and
+  continue working without having to learn a new format. More likely, even, is
+  that tooling will eventually support this transformation with a single
+  command.
+* Tooling that wishes to modify metadata on behalf of the user, for example,
+  package managers or dependency update automation in CI.
 
 
 Specification
 =============
 
 Any Python script may assign a variable named ``__pyproject__`` to a multi-line
-*double-quoted* string (``"""``) containing a valid TOML document. The opening
-of the string MUST be on the same line as the assignment. The closing of the
-string MUST be on a line by itself, and MUST NOT be indented.
+*double-quoted* string literal (``"""``) containing a valid TOML document. The
+opening of the string MUST be on the same line as the assignment. The closing
+of the string MUST be on a line by itself, and MUST NOT be indented.
+
+When there are multiple ``__pyproject__`` variables defined, tools MUST
+produce an error.
```

Comment:
Do you also want to disallow multiple assignment (``a = b = __pyproject__ = spam = """\..."""``)?

"same line as the assignment." The following is clearly banned, but which line *should* the ``"""`` appear on?

```python
__pyproject__\
    =\
        """...
"""
```

Reply:
Would there be a problem with specifying that it must start with a line matching exactly `__pyproject__ = """\n` (no empty space before)?

Reply:
I think as worded the specification prevents these pathological cases because the regular expression is documented as canonical which starts with a `^`, no?

Reply:
I made this explicit